python/llm/src/ipex_llm/transformers/models/llama.py

```diff
@@ -547,7 +547,7 @@ def llama_attention_forward_4_31_original(
             value_states = torch.cat(value_states, dim=-1)
         else:
             if fp16_fusion_check(self.q_proj, hidden_states, self.training) and \
-                    hidden_size == 4096:
+                    hidden_size == 4096 and self.k_proj.out_features != 512:
```

Comment:
I feel 512 is too specific ..
Would it be better to use `and self.q_proj.out_features == self.k_proj.out_features` ?